src/servers/src/query_handler/sql.rs

```diff
@@ -37,6 +38,13 @@ pub trait SqlQueryHandler {
         query_ctx: QueryContextRef,
     ) -> Vec<std::result::Result<Output, Self::Error>>;
 
+    async fn do_exec_plan(
+        &self,
+        query: &str,
+        plan: LogicalPlan,
+        query_ctx: QueryContextRef,
+    ) -> Vec<std::result::Result<Output, Self::Error>>;
```

Comment:
Do you still need this `query` string if plan provided?

Reply:
And if we are actually executing the plan, we can make the return type `Result<Output, Self::Error>` because plan only contains single query.